lib/mayaUsd/render/vp2RenderDelegate/points.cpp

```diff
@@ -143,19 +143,14 @@ void HdVP2Points::Sync(
     if (HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->normals)
         || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->widths)
         || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->primvar)) {
-        const HdVP2Material* material = static_cast<const HdVP2Material*>(
-            renderIndex.GetSprim(HdPrimTypeTokens->material, GetMaterialId()));
-
-        TfTokenVector        materialPrimvars;
-        const TfTokenVector* requiredPrimvars = &sFallbackShaderPrimvars;
-        TfToken              materialNetworkToken = _GetMaterialNetworkToken(reprToken);
-        if (material && material->GetSurfaceShader(materialNetworkToken)) {
-            materialPrimvars = material->GetRequiredPrimvars(materialNetworkToken);
-            materialPrimvars.push_back(HdTokens->widths);
-            requiredPrimvars = &materialPrimvars;
+        TfTokenVector requiredPrimvars;
+        if (!_GetMaterialPrimvars(renderIndex, GetMaterialId(), requiredPrimvars)) {
+            requiredPrimvars = sFallbackShaderPrimvars;
+        } else {
+            requiredPrimvars.push_back(HdTokens->widths);
```

Comment:
Since primvars (and their vertex buffers) are shared among all representations, we need to include here all required primvars from all available material networks, which is done inside _GetMaterialPrimvars method.